packages/expand-button-react/src/ExpandSection.tsx

```diff
@@ -0,0 +1,75 @@
+/* There are no types for core-toggle, so we have to use ts-ignore */
+// @ts-ignore: wait for nrk to supply types
+import CoreToggle from "@nrk/core-toggle/jsx";
+import cx from "classnames";
+import React, { ReactNode, useState } from "react";
+import { useAnimatedHeight } from "@fremtind/jkl-react-hooks";
+import type { ExpandButtonProps } from "./ExpandButton";
+import { ExpandButton } from "./ExpandButton";
+
+export interface ExpandSectionProps {
+    /**
+     * Innholdet som skal kunne ekspanderes og skjules.
+     * Om du ikke trenger at komponenten tar seg av visning og skjuling av
+     * innholdet skal du bruke `ExpandButton` direkte. Det sparer deg for
+     * å laste inn unødvendige avhengigheter.
+     */
+    children: ReactNode;
+    className?: string;
+    onClick?: (e: React.MouseEvent<HTMLButtonElement>, isExpanded: boolean) => void;
+    expandButtonProps?: Omit<ExpandButtonProps, "forceCompact" | "isExpanded" | "onClick" | "srOnly">;
+    title: string;
+    isExpanded?: boolean;
+    forceCompact?: boolean;
+}
+
+export const ExpandSection = ({
+    children,
+    className,
+    expandButtonProps,
+    isExpanded = false,
+    forceCompact,
+    onClick,
+    title,
+    ...rest
+}: ExpandSectionProps): JSX.Element => {
+    const [expanded, setIsExpanded] = useState(isExpanded);
+    const [elementRef] = useAnimatedHeight(expanded);
```

Comment:
`useAnimatedHeight` har mulighet for callbacks ved åpning/lukking via et `options`-argument. Kanskje det kunne eksponeres i Props?

https://github.com/fremtind/jokul/blob/1f3321e1db17432a5f299b891644acb34602a42e/packages/react-hooks/src/animation.tsx#L8